added disclaimer about network segments due to Serf limitations (#6558)
hashicorp/consul: website/source/docs/enterprise/network-segments/index.html.md

To get started with Network Segments,
[read the guide](https://learn.hashicorp.com/consul/day-2-operations/network-segments).

> -> **Note:** Due to limitations in Consul's implementation of Serf, you may currently only use 10 Network Segments per Consul cluster.

Reviewer: ```suggestion
-> **Note:** Due to limitations in [Serf](https://www.consul.io/docs/internals/gossip.html), you may currently only use up to 10 Network Segments per Consul cluster.
```

> -> **Note:** Due to limitations in [Serf](https://www.consul.io/docs/internals/gossip.html), you may currently only use up to 10 Network Segments per Consul cluster.

Reviewer 2: ```suggestion
-> **Note:** Due to limitations in [Serf](https://www.consul.io/docs/internals/gossip.html), you can only use up to 10 Network Segments per Consul cluster.
```
Did we decide if a Serf version number should be added?

> -> **Note:** Due to limitations in [Serf](https://www.consul.io/docs/internals/gossip.html), you may currently only use up to 10 Network Segments per Consul cluster.

Reviewer 3: ```suggestion
-> **Note:** Due to limitations in [Serf](https://www.consul.io/docs/internals/gossip.html), a Consul agent configured with too many Network Segments may not be able to start```

Reviewer 3: We should probably remove the arbitrary value of 10 Network Segments.

Reviewer 4: +1 if 10 isn't a real number we shouldn't say it, or we should hedge with "as few as" or something?
